src/processor/CodeSystemProcessor.ts

```diff
@@ -31,11 +32,40 @@ export class CodeSystemProcessor {
       ...CaretValueRuleExtractor.processResource(input, fisher, input.resourceType, config)
     );
     input.concept?.forEach((concept: any) => {
-      newRules.push(new ExportableConceptRule(concept.code, concept.display, concept.definition));
+      newRules.push(...this.extractConceptRules(concept, input, fisher));
     });
     target.rules = compact(newRules);
   }
 
+  static extractConceptRules(
+    concept: any,
+    codeSystem: ProcessableCodeSystem,
+    fisher: utils.Fishable,
+    hierarchy?: string[]
+  ): (ExportableCaretValueRule | ExportableConceptRule)[] {
+    const conceptRules: (ExportableConceptRule | ExportableCaretValueRule)[] = [];
+    const newConceptRule = new ExportableConceptRule(
+      concept.code,
+      concept.display,
+      concept.definition
+    );
+    newConceptRule.hierarchy = hierarchy ?? [];
+    concept._parentId = codeSystem.name ?? codeSystem.id;
+    concept._hierarchy = [...newConceptRule.hierarchy, concept.code];
```

Comment:
I think the other advantage of making a new `processConcept` function on the `CaretValueRuleExtractor` is that then you can just (I think) pass hierarchy and the parent ID as params, which makes things a little bit more readable.